assets/eip-4973/src/ERC4973.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.6;
+
+import {ERC165} from "openzeppelin-contracts/utils/introspection/ERC165.sol";
+import {Counters} from "openzeppelin-contracts/utils/Counters.sol";
+
+import {IERC721Metadata} from "./interfaces/IERC721Metadata.sol";
+import {IERC4973} from "./interfaces/IERC4973.sol";
+
+
+abstract contract ERC4973 is ERC165, IERC721Metadata, IERC4973 {
+  using Counters for Counters.Counter;
+
+  Counters.Counter private _tokenIds;
+  string private _name;
+  string private _symbol;
+
+  mapping(uint256 => address) private _bonds;
+  mapping(uint256 => string) private _tokenURIs;
+
+  constructor(
+    string memory name_,
+    string memory symbol_
+  ) {
+    _name = name_;
+    _symbol = symbol_;
+  }
+
+  function supportsInterface(bytes4 interfaceId) public view override returns (bool) {
+    return
+      interfaceId == type(IERC721Metadata).interfaceId ||
+      interfaceId == type(IERC4973).interfaceId ||
+      super.supportsInterface(interfaceId);
+  }
+
+  function name() public view virtual override returns (string memory) {
+    return _name;
+  }
+
+  function symbol() public view virtual override returns (string memory) {
+    return _symbol;
+  }
+
+  function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
+    require(_exists(tokenId), "tokenURI: token doesn't exist");
+    return _tokenURIs[tokenId];
+  }
+
+  function _exists(uint256 tokenId) internal view virtual returns (bool) {
+    return _bonds[tokenId] != address(0);
+  }
+
+  function ownerOf(uint256 tokenId) public view virtual returns (address) {
+    address owner = _bonds[tokenId];
+    require(owner != address(0), "ownerOf: token doesn't exist");
+    return owner;
+  }
+
+  function _mint(
```

Comment:
incremental `tokenId` increase is more opinionated than allowing it as a parameter/input